### DISCUSSION TITLE
2024.12.0bx Log Error

### DISCUSSION THREAD

**Discussion by Happy-Cadaver** [JurajNyiri/HomeAssistant-Tapo-Control#724] [General]
This could be me misunderstanding how things work.

The native TP-Link integration in 2024.12 beta discovered my cameras so I tried to add them and succeeded with my C520WS (my C100 failed but that's their bug) and it added 3 sensors, 2 of these overlap with Tapo Control... this may or may not be the cause.

Now both integrations log a warning/error, Tapo Control logs it as an error while TP-Link native logs it as a warning *shrug*.

It may also be that 2024.12.0 updates python to 3.13.

Do I add this as an issue to Tapo Control?

`This error originated from a custom integration.

Logger: custom_components.tapo_control
Source: custom_components/tapo_control/__init__.py:581
integration: Tapo: Cameras Control (documentation, issues)
First occurred: 12:17:04 PM (5 occurrences)
Last logged: 1:12:25 PM

Error: -40214, Response: {"error_code": -40214}
Error: Invalid stok value, Response: {"data": {"code": -40406}, "error_code": -40401}`

**Comment by JurajNyiri** [comment]
Do not use both integrations at the same time, choose one.

**Reply by Happy-Cadaver** [reply]
*grumble grumble growl bitch whine & moan*

I was afraid it would be that. Do you have any idea if this is because they are both on the same HAOS instance or because they are both polling the camera? I'd like to help them debug as they go, but I'm not giving up your level of control and feature set. Currently my 2024.11.3 instance shows no such logs, it's only my beta testing instance that does.

**Reply by JurajNyiri** [reply]
> they are both polling the camera

Camera can be polled by one integration / app at the same time.

That being said, this integration _should_ eventually recover, but if they keep fighting, none will gain control. I do not know what kind of logic they have for retries.

**Reply by Happy-Cadaver** [reply]
Thank you for the clarification, and of course all of the work that you do with this integration and the py-tapo project... you are appreciated. Especially for dealing with TP-Link directly to resolve the auth issue that's been plaguing us since April, thank you.
